src/routes/my-past-trips.js

```diff
@@ -0,0 +1,30 @@
+import * as utils from '../utils.js';
+
+export function get(req, res) {
+    const userId = req.user;
+    const now = new Date();
+
+    const leaderOnly = req.query.leaderOnly === 'true';
+
+    const tripsQuery = `
+      SELECT trips.id, title, location, start_time, end_time, description, leader,
```

Comment:
small nitpick, but I wouldn't capitalize functions, just because it makes them look like statements. A little indenting would help make this more readable too:

```
      SELECT 
        trips.id, title, location, start_time, end_time, description, leader,
        coalesce(clubs.name, 'None') as club
      FROM trip_members
      JOIN trips ON trips.id = trip_members.trip
      LEFT JOIN clubs ON trips.club = clubs.id
      WHERE 
        trip_members.user = ? 
        AND end_time < ?
        ${leaderOnly ? 'AND trip_members.leader = 1' : ''}
      ORDER BY end_time DESC
```